File: key_generator.js
const nacl = require('tweetnacl')
const base64 = require('base64-js')

// Generate an Ed25519 keypair
const keyPair = nacl.sign.keyPair()

// Convert keys to base64 strings
const private_key_base64 = base64.fromByteArray(keyPair.secretKey)
const public_key_base64 = base64.fromByteArray(keyPair.publicKey)

// Print keys in the base64 format
console.log("Private Key (Base64)")
console.log(private_key_base64)

console.log("Public Key (Base64):")
console.log(public_key_base64)

File: key-generator.py
import ed25519
import base64

# Generate an Ed25519 keypair
secret_key, public_key = ed25519.create_keypair()

# Convert keys to base64 strings
private_key_base64 = base64.b64encode(secret_key.to_bytes()).decode()
public_key_base64 = base64.b64encode(public_key.to_bytes()).decode()

# Print the keys in base64 format
print("Private Key (Base64):")
print(private_key_base64)

print("Public Key (Base64):")
print(public_key_base64)

